src/main/kotlin/org/stellar/walletsdk/Wallet.kt

```diff
@@ -111,4 +111,25 @@ class Wallet(
   fun removeAccountSigner(sourceAddress: String, signerAddress: String): Transaction {
     return addAccountSigner(sourceAddress, signerAddress, 0)
   }
+
+  // Submit transaction
+  fun submitTransaction(signedTransaction: Transaction, serverInstance: Server = server): Boolean {
+    try {
+      val response = serverInstance.submitTransaction(signedTransaction)
+
+      if (response.isSuccess) {
+        return true
+      }
+
+      var errorMessage = "Transaction failed"
+
+      if (!response.extras?.resultCodes?.transactionResultCode.isNullOrBlank()) {
+        errorMessage += ": ${response.extras.resultCodes.transactionResultCode}"
+      }
+
+      throw Exception(errorMessage)
```

Comment:
Let's take this out of the try/catch clause, since its not the code that may throw an exception.